napalm/base/test/getters.py

```diff
@@ -40,19 +40,36 @@ def dict_diff(prv, nxt):
     result = {}
 
     for k in keys:
-        if isinstance(prv.get(k), dict):
-            if isinstance(nxt.get(k), dict):
-                "If both are dicts we do a recursive call."
-                diff = dict_diff(prv.get(k), nxt.get(k))
+        a = prv.get(k)
+        b = nxt.get(k)
+
+        if isinstance(a, dict):
+            if isinstance(b, dict):
+                # If both are dicts we do a recursive call.
+                diff = dict_diff(a, b)
+                if diff:
+                    result[k] = diff
+            else:
+                # If only one is a dict they are clearly different
+                result[k] = {"result": a, "expected": b}
+        elif isinstance(a, list) and isinstance(b, list):
+            if any(isinstance(item, dict) for item in a) or any(
+                isinstance(item, dict) for item in b
+            ):
+                # Compare the lists of dictionaries
+                diff = list_dicts_diff(a, b)
                 if diff:
                     result[k] = diff
             else:
-                "If only one is a dict they are clearly different"
-                result[k] = {"result": prv.get(k), "expected": nxt.get(k)}
+                # Compare the lists values
+                # We sort them before comparing,
+                # otherwise lists in a different order won't match
+                if sorted(a) != sorted(b):
+                    result[k] = {"result": a, "expected": b}
```

Comment:
Why do we need the sorting--unit test data is static so the order should be the same? My concern is that we probably should be checking the order in the tests and this would be hiding that?

Reply:
It’s because of this:
https://travis-ci.org/napalm-automation/napalm/jobs/468657971#L1002

Every time I ran the test suite it failed in different ways. Somehow the order of the list changed on every run.

If you want, I’ll review it once more. It’s only after this change that I figured out how to fix debugging the unittests, so I can see more now.
This solution was more of a “add print statements untill it works” way of solving it.

Reply:
Reverted and root cause fixed.

```python
return [
    translation
    for entry, translation in C.LLDP_CAPAB_TRANFORM_TABLE.items()
    if entry in capabilities
]
```
Because dictionaries in python < 3.6 are not sorted, the result of the above piece of code isn't predictable.